lib/appsignal/hooks/active_job.rb

```diff
@@ -82,6 +82,12 @@ def execute(job)
               tags.merge(:status => :processed)
           end
         end
+
+        # sets the exception on the transaction, can be overridden by users
+        # to customize behavior. E.g. not sending the error on each occurrence.
+        def appsignal_set_error(transaction, exception)
```

Comment:
This is so you can override this method, right? In a subclass or monkeypatch.

I don't want to document that for other users to use it. I'm not against merging this, but it shouldn't be the way to go.

What you probably want is some kind of config option that allows you to configure after how many failed jobs it should report the error. Or automatically when it has exhausted its retries, it will report the error. But it won't report every retry.

Something like:
```yml
# config/appsignal.yml
production:
  activejob_only_report_after_all_retries: true
```

Reply:
> What you probably want is some kind of config option that allows you to configure after how many failed jobs it should report the error

Yes this is what we want. Preferably per Job instance. But that seemed quite a big change :-).

An alternative solution that might work and maybe more clean solution (so instead of needing to monkey patch or subclass) could be is:

```ruby
transaction.set_error(exception) unless respond_to?(:appsignal_skip_report_error?) && appsignal_skip_report_error?
```

Then in your job you could do:

```ruby
class SendInvoice < ApplicationJob
  def self.appsignal_skip_report_error? = true
```

What do you think?

Reply:
> Yes this is what we want.

Great!

> Preferably per Job instance. But that seemed quite a big change :-).

Ah.. yeah that would be more difficult to model in our config.

> An alternative solution that might work and maybe more clean solution (so instead of needing to monkey patch or subclass) could be is:

I like this approach 👍 But to avoid double negative thinking, we could call the method `appsignal_report_error?`, which is considered `true` if the method isn't found. If the method is present, returning `false` doesn't report the error.